<?xml version="1.0" encoding="UTF-8"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name="$(arg name)">

<link name="can_base_link">
    <inertial>
      <mass value="0.001"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
    </inertial>
  </link>

  
  <!-- Actual soup can link -->
  <link name="soup_can">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="0.349"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://isaac_manipulator_pick_and_place/urdf/meshes/tomato_soup.obj" scale="0.01 0.01 0.01"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://isaac_manipulator_pick_and_place/urdf/meshes/tomato_soup.obj" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
  </link>

  <!-- Fixed joint between the base link and the soup can -->
  <joint name="base_to_soup_can" type="fixed">
    <parent link="can_base_link"/>
    <child link="soup_can"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>
